x-pack/plugins/reporting/public/components/report_listing.tsx

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// TODO: Remove once typescript definitions are in EUI
+declare module '@elastic/eui' {
+  export const EuiBasicTable: React.SFC<any>;
+  export const EuiTextColor: React.SFC<any>;
+}
+
+import moment from 'moment';
+import React, { Component } from 'react';
+import chrome from 'ui/chrome';
+import { toastNotifications } from 'ui/notify';
+// @ts-ignore: implicit any for JS file
+import { Poller } from '../../../../common/poller';
+import { downloadReport } from '../lib/download_report';
+import { jobQueueClient } from '../lib/job_queue_client';
+import { ReportErrorButton } from './report_error_button';
+
+import {
+  EuiBasicTable,
+  EuiButtonIcon,
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiText,
+  EuiTextColor,
+  EuiTitle,
+  EuiToolTip,
+} from '@elastic/eui';
+
+interface Job {
+  id: string;
+  type: string;
+  object_type: string;
+  object_title: string;
+  created_by?: string;
+  created_at: string;
+  started_at?: string;
+  completed_at?: string;
+  status: string;
+  max_size_reached: boolean;
+}
+
+interface Props {
+  xpackInfo: any;
```

Comment:
What about passing in the three parameters used from the service?  Might make it easier to test too, and then you don't need `any`.  Like, `enableLinks`, `showLinks`, and `badLicenseMessage.